Spec/mastermind_spec.rb

```diff
@@ -23,6 +23,100 @@
   end
 end
 
+RSpec.describe Code, "compares guess with secret_key" do
+  let(:code) { Code.new }
+  let(:colors) { ["r", "g", "o", "y", "b", "p"] }
+  let(:key) { ["r", "r", "r", "r"] }
+  let(:input) { ["r", "r", "r", "r"] }
+
+  it "returns number of red pins" do
+    expect(code.red_counter(key, input)).to eq [1, 1, 1, 1]
+  end
+end
+
+RSpec.describe Code, "compares guess with secret_key" do
+  let(:code) { Code.new }
+  let(:colors) { ["r", "g", "o", "y", "b", "p"] }
+  let(:key) { ["r", "r", "y", "b"] }
+  let(:input) { ["r", "r", "p", "o"] }
+
+  it "returns number of red pins" do
+    expect(code.red_counter(key, input)).to eq [1, 1]
+  end
+end
+
+RSpec.describe Code, "trims code" do
+  let(:code) { Code.new }
+  let(:key) { ["r", "g", "y", "r"] }
+  let(:input) { ["r", "o", "b", "r"] }
+
+  it "deletes input position if it is the same as key index" do
+    expect(code.trim(key, input)).to eq ["", "o", "b", ""]
+  end
+end
+
+RSpec.describe Code, "trims code" do
+  let(:code) { Code.new }
+  let(:key) { ["y", "b", "p", "o"] }
+  let(:input) { ["y", "b", "p", "r"] }
+
+  it "deletes input position if it is the same as key index" do
+    expect(code.trim(key, input)).to eq ["", "", "", "r"]
+  end
+end
+
+RSpec.describe Code, "counts white pins" do
+  let(:code) { Code.new }
+  let(:key) { ["y", "b", "y", "r"] }
+  let(:input) { ["", "y", "b", ""] }
+
+  it "accurately counts white pins" do
```

Comment:
Like `describe #white_counter` would be the block within `RSpec.describe Code` that would have all tests related to that function`